packages/compass/src/main/auto-update-manager.ts

```diff
@@ -591,17 +623,9 @@ class CompassAutoUpdateManager {
       }
     });
 
-    compassApp.on('check-for-updates', () => {
-      this.setState(AutoUpdateManagerState.ManualCheck);
-    });
-
-    log.info(
-      mongoLogId(1001000133),
-      'AutoUpdateManager',
-      'Setting up updateManager',
-      { ...this.autoUpdateOptions, enabled }
-    );
-
+    // TODO(COMPASS-7233): This is kinda pointless at the moment because the
+    // preferences start as disabled and then become enabled the moment they are
+    // loaded. Which would immediately kick off the state change.
```

Comment:
I still think this is not a race condition and just a case for first-time compass run where some of those default properties are not set [until we show the welcome modal first time](https://github.com/mongodb-js/compass/blob/a23fde8b4d153893414b717423f0e48e00f69a36/packages/compass-home/src/components/home.tsx#L421-L427)